saleor/warehouse/models.py

```diff
@@ -96,10 +113,26 @@ def applicable_for_click_and_collect_no_quantity_check(
             return self._for_channel_click_and_collect(channel_id)
 
         stocks_qs = Stock.objects.filter(
-            product_variant__id__in=lines_qs.values("variant_id"),
-        ).select_related("product_variant")
+            product_variant_id__in=lines_qs.values("variant_id"),
+        )
 
-        return self._for_channel_lines_and_stocks(lines_qs, stocks_qs, channel_id)
+        number_of_variants = (
+            lines_qs.order_by("variant_id").distinct("variant_id").count()
+        )
+
+        warehouse_ids_with_stock_available = self._for_channel_lines_and_stocks(
+            number_of_variants, stocks_qs, channel_id
+        ).values("id")
+        lookup = Q(id__in=warehouse_ids_with_stock_available)
+        # if the stocks can cover the all variants, all C&C warehouses with
+        # `ALL_WAREHOUSES` option should be returned, as there is an option
+        # to ship the products to this point from another warehouse
+        stocks_count = len(set(stocks_qs.values_list("product_variant_id", flat=True)))
```

Comment:
Can you use the `count()` here as well?

Reply:
I cannot, the `product_variant_id` are duplicated, even when using `values("product_variant_id")`. 😞

Reply:
Cannot you use count with distinct?